baseapp/error.go

```diff
@@ -52,8 +59,18 @@ func HandleRouteError(w http.ResponseWriter, r *http.Request, err error) {
 	} else {
 		log = hlog.FromRequest(r).Error().Err(err)
 
+		cause := errors.Cause(err)
+		var errorMsg string
+		if aerr, ok := cause.(httpError); ok {
+			errorMsg = http.StatusText(aerr.StatusCode())
+		} else {
+			errorMsg = http.StatusText(http.StatusInternalServerError)
+		}
+
+		rid, _ := hlog.IDFromRequest(r)
 		WriteJSON(w, http.StatusInternalServerError, map[string]string{
```

Comment:
Need to replace `http.StatusInternalServerError` with the code from the `httpError`. Maybe extract the code on L62-68 and then get the message in one place using the code?

Reply:
Good point. I missed that in the refactor. I've set the status code to default to `http.StatusInternalServerError` and then overridden it if we get an actual code. As suggested I moved the error message generation down into one place too